typescript/src/redemption.ts

```diff
@@ -406,3 +409,79 @@ export async function getRedemptionRequest(
 
   return redemptionRequests[0]
 }
+
+/**
+ * Finds the oldest active wallet that has enough BTC to handle a redemption request.
+ * @param amount The amount to be redeemed.
+ * @param bridge The handle to the Bridge on-chain contract.
+ * @param bitcoinClient Bitcoin client used to interact with the network.
+ * @param bitcoinNetwork Bitcoin network.
+ * @returns Promise with the wallet details needed to request a redemption.
+ */
+export async function findWalletForRedemption(
+  amount: BigNumber,
+  bridge: Bridge,
+  bitcoinClient: BitcoinClient,
+  bitcoinNetwork: BitcoinNetwork
+): Promise<{
+  walletPublicKey: string
+  mainUTXO: UnspentTransactionOutput
+}> {
+  const wallets = await bridge.getNewWalletRegisteredEvents()
+
+  let walletData:
+    | {
+        walletPublicKey: string
+        mainUTXO: UnspentTransactionOutput
+      }
+    | undefined = undefined
+  let maxAmount = BigNumber.from(0)
+
+  for (const wallet of wallets) {
+    const prefixedWalletPublicKeyHash =
+      wallet.walletPublicKeyHash.toPrefixedString()
+    const { state, mainUtxoHash, walletPublicKey } = await bridge.wallets(
+      prefixedWalletPublicKeyHash
+    )
+
+    // Wallet must be in Live state.
+    if (state !== WalletState.Live) continue
+
```

Comment:
Let's also ensure the `mainUtxoHash` is non-zero before we proceed.

Reply:
96b1cfe